### ISSUE TITLE
tilde ~ not supported in g:vimwiki_listsym_rejected or g:vimwiki_listsyms

### ISSUE THREAD

**Issue by loganfred** [vimwiki/vimwiki#962]
Prior to submitting a new issue make sure to complete these steps:

- [x] Checkout the `dev` branch and confirm the issue is present there as well.
      The `dev` branch contains fixes that may not have been merged to `master` yet.
- [x] Post the syntax you are using (default/mediawiki/markdown) **and** your vimwiki settings from your `.vimrc`
```
$cd /tmp && git clone https://github.com/vimwiki/testwikis
```
```
" minimal .vimrc with offending line at end

call plug#begin('~/.vim/plugged')
Plug 'junegunn/vim-plug'
Plug 'vimwiki/vimwiki', {'branch': 'dev'}
call plug#end()

set runtimepath+=
set nocompatible
filetype plugin indent on
syntax enable

" default syntax
let vimwiki_default = {}
let vimwiki_default.path =  '/tmp/testwikis/vimwiki'
let vimwiki_default.path_html = ''
let vimwiki_default.syntax = 'default'
let vimwiki_default.ext = '.wiki'
let vimwiki_default.name = 'DefaultSyntax'

" markdown syntax - https://github.github.com/gfm/
let vimwiki_markdown = {}
let vimwiki_markdown.path =  '/tmp/testwikis/markdown'
let vimwiki_markdown.path_html = ''
let vimwiki_markdown.syntax = 'markdown'
let vimwiki_markdown.ext = '.md'
let vimwiki_markdown.name = 'MarkdownSyntax'

" mediawiki syntax - https://www.mediawiki.org/wiki/Help:Formatting
let vimwiki_mediawiki = {}
let vimwiki_mediawiki.path =  '/tmp/testwikis/mediawiki'
let vimwiki_mediawiki.path_html = ''
let vimwiki_mediawiki.syntax = 'mediawiki'
let vimwiki_mediawiki.ext = '.mw'
let vimwiki_mediawiki.name = 'MediaWikiSyntax'

" register the 3 wikis
let g:vimwiki_list = [vimwiki_default, vimwiki_markdown, vimwiki_mediawiki]

let g:vimwiki_listsym_rejected = '~'
```

default <leader>ww binding for <Plug>VimwikiIndex

- [x] Include the output of `:VimwikiShowVersion`.
```
Os: Mac
Vim: 802
Branch: dev
Revision: 7f8c4b5
Date: 2020-07-30 19:31:55 -0400
```

:messages
```
Error detected while processing function vimwiki#vars#init[2]..<SNR>11_populate_wikilocal_options[103]..<SNR>11_populate_list_vars:
line   28:
E33: No previous substitute regular expression
Error detected while processing function vimwiki#vars#get_wikilocal:
line    4:
E716: Key not present in Dictionary: rxListItem
```

Seems related to #194
 
**Not supported as global or local options in my testing.**

I didn't try any other exotic characters because I have no idea which characters to be immediately suspicious of as string inputs in Vim. Maybe it's worth a warning in the docs? This was a tough one to isolate.

**Comment by tinmarino** [comment]
Error in substitute => this variable is used in aregex => `:h /~` or `/~` and press enter will give you the same error message.
Escape it:
```vim
let g:vimwiki_listsym_rejected = '\~'
```

**Comment by loganfred** [comment]
Looks like trying to escape it gives a two character warning

```
Vimwiki Error: The provided value '\~' of the option 'g:vimwiki_listsym_rejected' must contain exactly 1 character(
s) but has 2. See ':h g:vimwiki__listsym_rejected'.
```

I'm not sure of the scope of this, but if it gets a good first issue tag maybe I can take a look.

I draw squiggly lines on abandoned to-do lists IRL so I like the tilde here.

**Comment by tinmarino** [comment]
Fixed: 708fde781afeaf

It could have been escaped with 
```vim
" Escape string to e matched as literal by a magic (normal) regex
" See :h pattern
function! vimwiki#u#escape_litteral(literal) abort
  return escape(a:literal, '$.*~\')
endfunction
```

But putting the regex in `[]` is what is commonly done in vimwiki

```diff
-  if match(a:wiki.listsyms, a:wiki.listsym_rejected) != -1
+  " Guard: Check if listym_rejected is in listsyms
+  if match(a:wiki.listsyms, '[' . a:wiki.listsym_rejected . ']') != -1
```
